client.go

```diff
@@ -256,6 +276,12 @@ func (c *Client) runOnce() error {
 	defer inputCh.Close()
 	defer outputCh.Close()
 
+	// Notify everyone that the client has started. Runs sequentially so there
```

Comment:
I think this should be placed just above the monitor or maybe even before the return of `nil` to ensure that if these functions gets called we can assume the connection is up, working and properly monitored.